src/Classes/GemSelectControl.lua

```diff
@@ -62,16 +62,21 @@ function GemSelectClass:PopulateGemList()
 	local showAll = self.skillsTab.showSupportGemTypes == "ALL"
 	local showAwakened = self.skillsTab.showSupportGemTypes == "AWAKENED"
 	local showNormal = self.skillsTab.showSupportGemTypes == "NORMAL"
+	local matchLevel = self.skillsTab.defaultGemLevel == "characterLevel"
+	local characterLevel = self.skillsTab.build and self.skillsTab.build.characterLevel or 1
 	for gemId, gemData in pairs(self.skillsTab.build.data.gems) do
-		if (showAwakened or showAll) and gemData.grantedEffect.plusVersionOf then
-			self.gems["Default:" .. gemId] = gemData
-		elseif showNormal or showAll then
-			if self.skillsTab.showAltQualityGems and self.skillsTab.defaultGemQuality or 0 > 0 then
-				for _, altQual in ipairs(self.skillsTab:getGemAltQualityList(gemData)) do
-					self.gems[altQual.type .. ":" .. gemId] = gemData
-				end
-			else
+		local levelRequirement = gemData.grantedEffect.levels[1].levelRequirement or 1
```

Comment:
This needs to be called on the appropriate gem level otherwise it breaks see my original change.

Reply:
Where/how exactly does it break? ``gemData.grantedEffect.levels[1].levelRequirement`` should always return the level requirement for the first (aka lowest) entry in the gem data, which is all we care about for filtering out gems where the minimum level requirement is too high for the character to equip.

Reply:
You are correct I think I was getting confused due to the caching issues.